pandas/core/groupby/groupby.py

```diff
@@ -1284,10 +1293,18 @@ def first(x):
                 return first(x)
 
         def last_compat(x, axis=0):
+            # This is a bit strange.
+            # We only hit this block when grouping a DatetimeTZBlock *and*
+            # a categorical. Something strange going on with first for
+            # categorical dta.
```

Comment:
xref #21603

I think the goal is to eventually rewrite `first` and `last` in terms of `nth`